utilities/pipelines/resourcePublish/Publish-ModuleToPrivateBicepRegistry.ps1

```diff
@@ -63,88 +58,34 @@ function Publish-ModuleToPrivateBicepRegistry {
 
-        if (-not ($repositories = Get-AzContainerRegistryRepository -RegistryName $bicepRegistryName -ErrorAction 'SilentlyContinue')) {
-            # No repositories yet
-            $repositories = @()
-        }
-
-        if ($repositories.Contains($moduleRegistryIdentifier)) {
-            $versions = (Get-AzContainerRegistryTag -RegistryName $bicepRegistryName -RepositoryName $moduleRegistryIdentifier).Tags.Name
-            $latestVersion = (($versions -as [Version[]]) | Measure-Object -Maximum).Maximum
-            Write-Verbose "Published versions detected in private bicep registry [$moduleIdentifier]. Fetched latest [$latestVersion]."
-        } else {
-            Write-Verbose "No version for module reference [$moduleRegistryIdentifier] detected in private bicep registry [$bicepRegistryName]. Creating new."
-            $latestVersion = New-Object System.Version('0.0.0')
-        }
-
-        ############################
-        ##    EVALUATE VERSION    ##
-        ############################
-
-        if (-not ([String]::IsNullOrEmpty($customVersion)) -and ((New-Object System.Version($customVersion)) -gt (New-Object System.Version($latestVersion)))) {
-            Write-Verbose "A custom version [$customVersion] was specified in the pipeline script and is higher than the current latest. Using it."
-            $newVersion = $customVersion
-        } else {
-            Write-Verbose 'No custom version set. Using default versioning.'
-
-            switch ($versioningOption) {
-                'major' {
-                    Write-Verbose 'Apply version update on "major" level'
-                    $newVersion = (New-Object -TypeName System.Version -ArgumentList ($latestVersion.Major + 1), 0, 0).ToString()
-                    break
-                }
-                'minor' {
-                    Write-Verbose 'Apply version update on "minor" level'
-                    $newVersion = (New-Object -TypeName System.Version -ArgumentList $latestVersion.Major, ($latestVersion.Minor + 1), 0).ToString()
-                    break
-                }
-                'patch' {
-                    Write-Verbose 'Apply version update on "patch" level'
-                    $newVersion = (New-Object -TypeName System.Version -ArgumentList $latestVersion.Major, $latestVersion.Minor, ($latestVersion.Build + 1)).ToString()
-                    break
-                }
-                default {
-                    throw "Unsupported version option: $versioningOption."
-                }
-            }
-        }
-
-        $newVersionObject = New-Object System.Version($newVersion)
-        if ($newVersionObject -lt $latestVersion -or $newVersionObject -eq $latestVersion) {
-            throw ('The provided custom version [{0}] must be higher than the current latest version [{1}] published in the private bicep registry [{2}]' -f $newVersionObject.ToString(), $latestVersion.ToString(), $bicepRegistryName)
```

Comment:
In the past we thre an exception here which we agreed upon not doing anymore. I'm however wondering: Did you do a test and check what happens if we publish an 'older' version to all the target artifact stores?
I'm especially interested if all locations are able to identify that one tries to publish a lower version than the latest published (not that we also have a name suffix)

Reply:
Doing a test for all 3 target publishing platforms now.

Reply:
As proven by a full rerun of both ADO pipelines and GH workflows for Analysis Services Servers, it does indeed support publish of a lower version, also a overwrite (by the looks of it) as it caused no errors for the deployments when run after each other with about 5 min delay.

Created an issue to improve this: #941